cassiopeia/simulator/Cas9LineageTracingDataSimulator.py

```diff
@@ -132,42 +144,70 @@ def __init__(
         self.size_of_cassette = size_of_cassette
         self.number_of_cassettes = number_of_cassettes
         self.collapse_sites_on_cassette = collapse_sites_on_cassette
+        number_of_characters = size_of_cassette * number_of_cassettes
 
         if isinstance(mutation_rate, float):
             if mutation_rate < 0:
                 raise DataSimulatorError(
                     "Mutation rate needs to be" " non-negative."
                 )
-            number_of_characters = size_of_cassette * number_of_cassettes
             self.mutation_rate_per_character = [
                 mutation_rate
             ] * number_of_characters
-        else:
-            if len(mutation_rate) != (
-                self.number_of_cassettes * self.size_of_cassette
-            ):
+        elif isinstance(mutation_rate, list):
```

Comment:
might be cleaner just to cast `mutation_rate` to `list` rather than throw an error if `mutation_rate` is a numpy array? what do you think?